boa/src/vm/mod.rs

```diff
@@ -5,25 +5,44 @@ pub(crate) mod instructions;
 
 pub use compilation::Compiler;
 pub use instructions::Instruction;
+use std::time::{Duration, Instant};
 
 // Virtual Machine.
 #[derive(Debug)]
 pub struct VM<'a> {
     ctx: &'a mut Context,
+    idx: usize,
     instructions: Vec<Instruction>,
     pool: Vec<Value>,
     stack: Vec<Value>,
     stack_pointer: usize,
+    profile: Profiler,
+    is_trace: bool,
+}
+#[derive(Debug)]
+pub struct Profiler {
+    instant: Instant,
+    prev_time: Duration,
+    trace_string: String,
+    start_flag: bool,
 }
 
 impl<'a> VM<'a> {
-    pub fn new(compiler: Compiler, ctx: &'a mut Context) -> Self {
+    pub fn new(compiler: Compiler, ctx: &'a mut Context, _trace: bool) -> Self {
         Self {
             ctx,
+            idx: 0,
             instructions: compiler.instructions,
             pool: compiler.pool,
             stack: vec![],
             stack_pointer: 0,
+            is_trace: true,
```

Comment:
I'm confused, isn't `_trace` meant to be the value used to initialize `is_trace`?

Reply:
Yes you’re right, I had some issues there I need to sort that out. I left that open because I had a question if we’re happy to have all the the new() calls have trace as an argument

Reply:
This has been resolved